scripts/iSnobal/environment.yml

```diff
@@ -0,0 +1,494 @@
+  - prompt-toolkit=3.0.42
+  - prompt_toolkit=3.0.42
+  - psutil=5.9.8
+  - pthread-stubs=0.4
+  - ptyprocess=0.7.0
+  - pugixml=1.14
+  - pulseaudio-client=16.1
+  - pure_eval=0.2.2
+  - py-opencv=4.9.0
+  - pyarrow=15.0.0
+  - pyarrow-hotfix=0.6
+  - pycparser=2.21
+  - pygments=2.17.2
+  - pyparsing=3.1.2
+  - pyproj=3.6.1
+  - pyqt=5.15.9
+  - pyqt5-sip=12.12.2
+  - pyshp=2.3.1
+  - pysocks=1.7.1
+  - python=3.12.2
+  - python-dateutil=2.9.0
+  - python-fastjsonschema=2.19.1
+  - python-json-logger=2.0.7
+  - python-tzdata=2024.1
+  - python_abi=3.12
+  - pytz=2024.1
+  - pyviz_comms=3.0.1
+  - pywavelets=1.4.1
+  - pyyaml=6.0.1
+  - pyzmq=25.1.2
+  - qt-main=5.15.8
+  - qtconsole-base=5.5.1
+  - qtpy=2.4.1
+  - rasterio=1.3.9
+  - rav1e=0.6.6
+  - rdma-core=50.0
+  - re2=2023.09.01
+  - readline=8.2
+  - referencing=0.34.0
+  - requests=2.31.0
+  - rfc3339-validator=0.1.4
+  - rfc3986-validator=0.1.1
+  - rioxarray=0.15.1
+  - rpds-py=0.18.0
+  - rtree=1.2.0
+  - s2n=1.4.5
+  - scikit-image=0.22.0
+  - scikit-learn=1.4.1.post1
+  - scipy=1.12.0
+  - seaborn=0.13.2
+  - seaborn-base=0.13.2
+  - send2trash=1.8.2
+  - setuptools=69.2.0
+  - shapely=2.0.3
+  - sip=6.7.12
+  - six=1.16.0
+  - snappy=1.1.10
+  - sniffio=1.3.1
+  - snuggs=1.4.7
+  - sortedcontainers=2.4.0
+  - soupsieve=2.5
+  - sqlite=3.45.2
+  - stack_data=0.6.2
+  - statsmodels=0.14.1
+  - svt-av1=1.8.0
+  - tbb=2021.11.0
+  - tblib=3.0.0
+  - terminado=0.18.1
+  - threadpoolctl=3.4.0
+  - tifffile=2024.2.12
+  - tiledb=2.20.1
+  - tinycss2=1.2.1
+  - tk=8.6.13
+  - tmux=3.4
+  - toml=0.10.2
+  - tomli=2.0.1
+  - toolz=0.12.1
+  - tornado=6.4
+  - tqdm=4.66.2
+  - traitlets=5.14.2
+  - traittypes=0.2.1
+  - types-python-dateutil=2.9.0.20240316
+  - typing-extensions=4.10.0
+  - typing_extensions=4.10.0
+  - typing_utils=0.1.0
+  - tzcode=2024a
+  - tzdata=2024a
+  - uc-micro-py=1.0.3
+  - ucx=1.15.0
+  - uri-template=1.3.0
+  - uriparser=0.9.7
+  - urllib3=2.2.1
+  - wcwidth=0.2.13
+  - webcolors=1.13
+  - webencodings=0.5.1
+  - websocket-client=1.7.0
+  - wheel=0.42.0
+  - widgetsnbextension=4.0.10
+  - x264=1!164.3095
+  - x265=3.5
+  - xarray=2024.2.0
+  - xcb-util=0.4.0
+  - xcb-util-image=0.4.0
+  - xcb-util-keysyms=0.4.0
+  - xcb-util-renderutil=0.3.9
+  - xcb-util-wm=0.4.1
+  - xerces-c=3.2.5
+  - xkeyboard-config=2.41
+  - xorg-fixesproto=5.0
+  - xorg-inputproto=2.3.2
+  - xorg-kbproto=1.0.7
+  - xorg-libice=1.1.1
+  - xorg-libsm=1.2.4
+  - xorg-libx11=1.8.7
+  - xorg-libxau=1.0.11
+  - xorg-libxdmcp=1.1.3
+  - xorg-libxext=1.3.4
+  - xorg-libxfixes=5.0.3
+  - xorg-libxi=1.7.10
+  - xorg-libxrender=0.9.11
+  - xorg-renderproto=0.11.1
+  - xorg-xextproto=7.3.0
+  - xorg-xf86vidmodeproto=2.3.1
+  - xorg-xproto=7.0.31
+  - xyzservices=2023.10.1
+  - xz=5.2.6
+  - yaml=0.2.5
+  - zeromq=4.3.5
+  - zfp=1.0.1
+  - zict=3.0.0
+  - zipp=3.17.0
+  - zlib=1.2.13
+  - zlib-ng=2.0.7
+  - zstd=1.5.5
+  - pip:
+      - jupyterlab==4.2.2
+      - jupyterlab-server==2.27.2
```

Comment:
Over the years I adapted to use:
```
conda env export --from-history
```
to create environment yaml files. Using only the `export` gives you very specific package versions and all it's dependencies (as seen above) and will make it hard to re-use this file later. The `--from-history` are only the packages that you installed and makes this list way more compact. Afterwards, you can go in and pin down the specific version you used.

Additionally, I also try to not include any bigger environments like `jupyterlab` and rather have this environment be ready as an additional kernel via `ipython`. This again reduces the list of needed packages and makes it easier to resolve for `conda` when creating one from scratch.

Reply:
Got it, thanks. Updating the yaml now